src/rgw/rgw_common.h

```diff
@@ -1435,6 +1436,11 @@ class RGWEnv;
 /* Namespaced forward declarations. */
 namespace rgw {
   namespace auth {
+    class Identity;
+    // size of static storage needed to fit any derived Identity
+    static constexpr size_t MaxIdentitySize = 704;
```

Comment:
Wow, pretty large value. I would love to have the possibility to delegate the computations to compiler. IIRC in the initial implementation of `static_ptr` there was a template to get maximal `sizeof` among its arg pack. Though, this approach could require extensive code shuffling in headers. :-(